lib/cli/clients/ega/ega_script.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/python
+import subprocess
+import random
+import string
+import logging
+
+
+def ega_call(id, passfile, toolpath, udt, downloaddirectory):
+    logger = logging.getLogger('__log__')
+
+    lable = id + '_request'
+    key = ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(4))
```

Comment:
Provided you don't have any name clashes and the intention is clear it is usually better to do this:
```python
from random import SystemRandom
from string import ascii_uppercase, digits
```

because then your code would look like this:
```python
''.join(SystemRandom().choice(ascii_uppercase + digits) for _ in range(4))
```